### PR TITLE
Clarification on containers immutability

### LINKED ISSUE #37959: Clarify what means immutable by design
**Problem:**

I find this information confusing:

https://github.com/kubernetes/website/blob/9ac44f2f332144d87499b6b610b45b7bba252bad/content/en/docs/concepts/containers/_index.md?plain=1#L34-L37

Containers are designed to be staless and immutable, but it is the responsibility of the users to maintain these properties. The wording "by design" suggests that the container is immutable however it is used.

It is incorrect that you cannot  change the code of a container that is already running. I have seen the anti-pattern of running `apt-get update && apt-get install <package>` in the containers entrypoint. This makes the container not reproducible, because of the download of a different version of the software every time the container starts, and it proves that the container "by design" is not immutable.

More on this container operations thoughts can be found here:

https://cloud.google.com/architecture/best-practices-for-operating-containers#ensure_that_your_containers_are_stateless_and_immutable

**Proposed Solution:**

I will propose a PR with the following:

Containers are designed to be stateless and immutable: you should not change the code of a 
container that is already running. If you have a containerized application 
 and want to make changes, the correct process is to build a new image that includes 
 the change, then recreate the container to start from the updated image. 

**Page to Update:**
https://kubernetes.io/docs/concepts/containers/

### PR DESCRIPTION
**Pull request by zioproto**
Clarification on container immutability.

Fixes #37959 

References: https://cloud.google.com/architecture/best-practices-for-operating-containers#ensure_that_your_containers_are_stateless_and_immutable

### CODE REVIEW

File: content/en/docs/concepts/containers/_index.md [hunk 1]
```diff
@@ -31,10 +31,11 @@ software package, containing everything needed to run an application:
 the code and any runtime it requires, application and system libraries,
 and default values for any essential settings.
 
-By design, a container is immutable: you cannot change the code of a
-container that is already running. If you have a containerized application
-and want to make changes, you need to build a new image that includes
-the change, then recreate the container to start from the updated image.
+Containers are designed to be stateless and immutable: you should not change
```
Comment (sftim) [inline content/en/docs/concepts/containers/_index.md:34]:
```suggestion
Containers are intended to be stateless and
[immutable](https://glossary.cncf.io/immutable-infrastructure/):
you should not change
```

**Review by sftim — COMMENTED** [review]
/lgtm

I'd write "intended" rather than "designed"

**Comment by zioproto** [comment]
@sftim thanks for your review. I accepted your suggested change. It lgtm I have nothing more to add.

File: content/en/docs/concepts/containers/_index.md [hunk 2]
```diff
@@ -31,10 +31,13 @@ software package, containing everything needed to run an application:
...
-By design, a container is immutable: you cannot change the code of a
-container that is already running. If you have a containerized application
-and want to make changes, you need to build a new image that includes
-the change, then recreate the container to start from the updated image.
+Containers are intended to be stateless and
+[immutable](https://glossary.cncf.io/immutable-infrastructure/):
+you should not change
+the code of a container that is already running. If you have a containerized
+application and want to make changes, the correct process is to build a new
+image that includes the change, then recreate the container to start from the
```
Comment (sftim) [inline content/en/docs/concepts/containers/_index.md:39]:
```suggestion
image that includes the change, then recreate the container (on Kubernetes,
this means starting a replacement Pod) with a definition that runs the
```

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm
/approve

**Pull request merged**